spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/core/PubSubTemplate.java

```diff
@@ -132,15 +133,20 @@ else if (payload instanceof byte[]) {
 			pubsubMessageBuilder.setData(ByteString.copyFrom((byte[]) payload));
 		}
 		else {
-			pubsubMessageBuilder.setData(
-					ByteString.copyFrom(this.messageConverter.toPayload(payload)));
+			try {
+				pubsubMessageBuilder.setData(
+						ByteString.copyFrom(this.messageConverter.toPayload(payload)));
+			}
+			catch (IOException ioe) {
+				throw new UncheckedIOException("Error serializing payload into JSON. ", ioe);
```

Comment:
`into JSON`?

Are you sure that `messageConverter` is always about JSON?

Can't I inject any other possible `MessageConverter` impl even if JSON is a default one?